### PR TITLE
Add TPCDS query output schema check

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Use TPCDS query to check if we have some behavior change.

Note that 
1. this PR only add `q1`.
2. this PR just to check the output schema. The output data check can be added in future.
3. this new added test file structure looks like
```
tpcds-1_4
  - q1
    - q1.sql
    - q1.output.schema 
    - q1.output.data (if we support this in future)
  - q2
  ......
```

### _How was this patch tested?_
Add new test

### CODE REVIEW

File: kyuubi-main/src/test/resources/tpcds-1_4/q1/q1.output.schema [hunk 1]
```diff
@@ -0,0 +1 @@
+struct<c_customer_id:string>
```
Comment (turboFei) [inline kyuubi-main/src/test/resources/tpcds-1_4/q1/q1.output.schema:1]: nit: newline

Comment (ulysses-you) [reply kyuubi-main/src/test/resources/tpcds-1_4/q1/q1.output.schema]: add `\n` when generate golden file.

File: kyuubi-main/src/test/resources/log4j.properties [hunk 2]
```diff
@@ -32,7 +32,7 @@ log4j.appender.FA.layout=org.apache.log4j.PatternLayout
 log4j.appender.FA.layout.ConversionPattern=%d{HH:mm:ss.SSS} %t %p %c{2}: %m%n
 
 # Set the logger level of File Appender to WARN
-log4j.appender.FA.Threshold = DEBUG
+log4j.appender.FA.Threshold = INFO
```
Comment (ulysses-you) [inline kyuubi-main/src/test/resources/log4j.properties:35]: make log small if we add tpcds check

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/tpcds/TPCDSOutputSchemaSuite.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  override def afterAll(): Unit = {
+    withJdbcStatement() { statement =>
+      statement.execute(s"DROP DATABASE IF EXISTS $database CASCADE")
+    }
+    super.afterAll()
+  }
+
+  private val regenerateGoldenFiles = sys.env.get("KYUUBI_GENERATE_GOLDEN_FILES").contains("1")
```
Comment (yaooqinn) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/operation/tpcds/TPCDSOutputSchemaSuite.scala:61]: we can reuse KYUUBI_UPDATE here, it's much shorter

**Comment by yaooqinn** [comment]
Great work, this is a huge step!

**Comment by ulysses-you** [comment]
Travis passed. 

The GA failure is not related 
```
Error:  ## Exception when compiling 13 sources to /home/runner/work/kyuubi/kyuubi/externals/kyuubi-spark-sql-engine/target/scala-2.12/test-classes
```

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/tpcds/TPCDSOutputSchemaSuite.scala [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+          q.getFileName.toString,
+          s"${q.getFileName}.output.schema"
+        )
+        val queryString = fileToString(queryFile)
+        runQuery(queryString, schemaFile)
+      }
+    }
+  }
+
+  runQueries("tpcds-1_4")
```
Comment (turboFei) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/operation/tpcds/TPCDSOutputSchemaSuite.scala:117]: could we refer https://github.com/apache/spark/blob/7242d7f774b821cbcb564c8afeddecaf6664d159/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L220-L244 to make it automatically?

Comment (turboFei) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/operation/tpcds/TPCDSOutputSchemaSuite.scala:117]: could we refer https://github.com/apache/spark/blob/7242d7f774b821cbcb564c8afeddecaf6664d159/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L238-L243 to create test for each query?

**Review by turboFei — APPROVED** [review]
LGTM, thanks

**Comment by turboFei** [comment]
```
Error:  Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.3.0:testCompile (scala-test-compile-first) on project kyuubi-spark-sql-engine: wrap: scala.reflect.internal.FatalError: Error accessing /home/runner/.m2/repository/org/apache/hive/hive-llap-common/2.3.7/hive-llap-common-2.3.7.jar: error in opening zip file -> [Help 1]
```

**Review by yaooqinn — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merging to master

**Comment by turboFei** [comment]
seems there is some issue for reopened a pr.

I will use github merge function to merge it. 

Thanks
image

**Pull request merged**